vtysh/vtysh_main.c

```diff
@@ -424,9 +431,9 @@ int main(int argc, char **argv, char **env)
 			"most one of:\n\t-b, -C, -m, -w\n");
 		return 1;
 	}
-	if (inputfile && (writeconfig || boot_flag)) {
+	if (inputfile && (writeconfig || boot_flag || preconfig)) {
```

Comment:
I think we will print this everytime when -p option is given. With -p options we set inputfile and also flag preconfig. I guess we need to explicitly set a flag for -f such that we use that instead of inputfile in conditional check.